src/hatch_egg_task.c

```diff
@@ -0,0 +1,100 @@
+#include "hatch_egg_task.h"
+
+#include "get_egg.h"
+#include "naming_screen.h"
+#include "trainer_memo.h"
+#include "unk_02055244.h"
+#include "unk_020552A4.h"
+
+typedef struct HatchEggTaskData {
+    u32 state;
+    BOOL wantToNameNewMon;
+    NamingScreenArgs *namingScreenArgs;
+    UnkStruct_02091240 unkC;
+} HatchEggTaskData;
+
+FS_EXTERN_OVERLAY(OVY_95);
+extern BOOL ov95_021E6D70(OVY_MANAGER *, int *);
+extern BOOL ov95_021E6E88(OVY_MANAGER *, int *);
+extern BOOL ov95_021E6F0C(OVY_MANAGER *, int *);
+
+static const OVY_MGR_TEMPLATE _02106048 = {
+    .init   = ov95_021E6D70,
+    .exec   = ov95_021E6E88,
+    .exit   = ov95_021E6F0C,
+    .ovy_id = FS_OVERLAY_ID(OVY_95),
+};
+
+static BOOL Task_HatchEggInParty(TaskManager *taskManager) {
+    HatchEggTaskData *data = TaskManager_GetEnvironment(taskManager);
+
+    switch (data->state) {
+    case 0: // Leave overworld
+        sub_0206D328(data->unkC.mon, HEAP_ID_FIELD);
+        CallTask_LeaveOverworld(taskManager);
+        data->state++;
+        break;
+    case 1: // Open the egg hatch app
+        CallApplicationAsTask(taskManager, &_02106048, data);
+        data->state++;
+        break;
+    case 2: { // Update egg status, add in mapsec for hatch location, update Pokedex if necessary
+        FieldSystem *fieldSystem = TaskManager_GetFieldSystem(taskManager);
+        Pokemon *pokemon         = data->unkC.mon;
+        PlayerProfile *profile   = Save_PlayerData_GetProfileAddr(FieldSystem_GetSaveData(fieldSystem));
+
+        u32 mapsec = MapHeader_GetMapSec(fieldSystem->location->mapId);
+        u32 isEgg  = 0;
```

Comment:
Should be a BOOL?

Reply:
There was one spot in `sub_0206D328` where this value was set to 70. For now I'll make it a BOOL